tests/epyccel/test_return.py

```diff
@@ -0,0 +1,140 @@
+        return ((4 + 5)/(6 - 3) * a[0])%(9 - a[1])
+
+    def multi_allocs(i :int):
+        import numpy as np
+        a = np.ones(i)
+        b = np.ones(i)
+        c = np.ones(i)
+        d = np.ones(i)
+        e = np.ones(i)
+        return ((4 + 5)/(d[0] + e[2]) * c[0])%(b[2] + a[1]) - 4
+
+    # these flags are used to make sure that Pyccel doesn't print an unused variable generated from a return statement
+    flags = "-Werror -Wunused-variable"
+
+    epyc_single_return_var_assign = epyccel(single_return_var_assign, language=language)
+    assert (epyc_single_return_var_assign() == single_return_var_assign())
+
+    epyc_assign_vars_return = epyccel(assign_vars_return, language=language, fflags=flags)
+    assert (epyc_assign_vars_return(3, 4) == assign_vars_return(3, 4))
+
+    epyc_sum_in_single_return = epyccel(sum_in_single_return, language=language, fflags=flags)
+    assert (epyc_sum_in_single_return(7, 2) == sum_in_single_return(7, 2))
+
+    epyc_return_expr = epyccel(return_expr, language=language, fflags=flags)
+    assert (epyc_return_expr(7, 2) == return_expr(7, 2))
+
+    epyc_return_single_var = epyccel(return_single_var, language=language, fflags=flags)
+    assert (epyc_return_single_var(7) == return_single_var(7))
+
+    epyc_return_scalare = epyccel(return_scalare, language=language, fflags=flags)
+    assert (epyc_return_scalare() == return_scalare())
+
+    epyc_multi_return_scalare = epyccel(multi_return_scalare, language=language, fflags=flags)
+    assert (epyc_multi_return_scalare() == multi_return_scalare())
+
+    epyc_multi_return_vars = epyccel(multi_return_vars, language=language, fflags=flags)
+    assert (epyc_multi_return_vars(7, 2) == multi_return_vars(7, 2))
+
+    epyc_multi_return_vars_expr = epyccel(multi_return_vars_expr, language=language, fflags=flags)
+    assert (epyc_multi_return_vars_expr(7, 2) == multi_return_vars_expr(7, 2))
+
+    epyc_scalare_multi_return_stmts = epyccel(scalare_multi_return_stmts, language=language, fflags=flags)
+    assert (epyc_scalare_multi_return_stmts(7) == scalare_multi_return_stmts(7))
+
+    epyc_create_arr = epyccel(create_arr, language=language, fflags=flags)
+    assert (epyc_create_arr(7) == create_arr(7))
+
+    epyc_return_arr_element = epyccel(return_arr_element, language=language, fflags=flags)
+    assert (epyc_return_arr_element(7) == return_arr_element(7))
+
+    epyc_create_multi_arrs = epyccel(create_multi_arrs, language=language, fflags=flags)
+    assert (epyc_create_multi_arrs(7) == create_multi_arrs(7))
+
+    epyc_expr_arrs_elements = epyccel(expr_arrs_elements, language=language, fflags=flags)
+    assert (epyc_expr_arrs_elements(7) == expr_arrs_elements(7))
+
+    epyc_complex_expr = epyccel(complex_expr, language=language, fflags=flags)
+    assert (epyc_complex_expr(7) == complex_expr(7))
+
+    epyc_multi_allocs = epyccel(multi_allocs, language=language, fflags=flags)
+    assert (epyc_multi_allocs(7) == multi_allocs(7))
```

Comment:
Please split this into multiple functions. This helps for a couple of reasons:
1. It is easier to see which method raised the error
2. We can get information about all failures at once instead of just the first failure. If we only wanted the first failure we would have run pytest with the `-x` option